### PR TITLE
Add client_api example

### PR DESCRIPTION
**Pull request by YuanTingHsieh**
### Description

- Adds PTFilePipeLauncherExecutor, NumpyToPTParamsConverter, PTToNumpyParamsConverter
- Adds an example to showcase how SubprocessLauncher + PTFilePipeLauncherExecutor + nvflare client interface with pytorch training script.
   We choose CIFAR10 (one of the simplest application) to showcase this.

Depends on #1863 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py [hunk 1]
```diff
@@ -106,11 +132,14 @@ def main():
...
 
     trainer = Cifar10Trainer(args.dataset_root, lr, epochs)
+    global_metrics = trainer.evaluate(weights=input_weights)
     trainer.train(weights=input_weights)
 
     output_params = {k: v.cpu().numpy() for k, v in trainer.model.state_dict().items()}
     # send trained model back to NVFlare
-    output_fl_model = FLModel(params=output_params, params_type=ParamsType.FULL)
+    output_fl_model = FLModel(
+        params=output_params, params_type=ParamsType.FULL, meta={MetaKey.INITIAL_METRICS: global_metrics}
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py:141]: The FLModel will be translated/converted into a DXO (for compatible with old server side component).
As InTimeModelSelector (https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/app_common/widgets/intime_model_selector.py#L36) required this metric to be put inside DXO's meta with key MetaKey.INITIAL_METRICS.

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py:141]: This is different from if we use FLModel to represents a "metric" for cross-site validation, in that case we will do
`FLModel(metrics=xxxx)`

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py:141]: I can add client_weights

[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py]: Add some missing concepts per discussion, will send out an email to team to discuss these changes.

File: examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    nvflare.init(data_exchange_path=args.data_exchange_root)
+
+    # get model from NVFlare
+    input_weights = nvflare.receive_model()
+
+    trainer = Cifar10Trainer(args.dataset_root, lr, epochs)
+    global_metrics = trainer.evaluate(weights=input_weights)
+    trainer.train(weights=input_weights)
+
+    nvflare.submit_model(model=trainer.model.state_dict(), initial_metrics=global_metrics)
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/jobs/subprocess-file/app_client/custom/cifar10.py]: changed.

File: examples/advanced/loose-integration/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Loose Integration with NVFlare
+
+In this directory, we are providing job configurations to showcase how to utilizes 
+`LauncherExecutor`, `Launcher` and `ModelExchanger` to complete loose integration with NVFlare.
+
+
+## Examples
+
+- Use `SubprocessLauncher`, `FilePipeLauncherExecutor` and `FilePipeModelExchanger` to
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/README.md]: added more info.

File: nvflare/app_common/model_exchange/__init__.py [hunk 4]
```diff
@@ -11,3 +11,51 @@
...
+PTModelConverter, pt_import_ok = optional_import(module="nvflare.app_opt.pt.model_converter", name="PTModelConverter")
+
+if pt_import_ok:
+    format_dict.update({"pytorch": PTModelConverter()})
+
+
+process_dict: Dict[int, Tuple[ModelConverter, FilePipeModelExchanger]] = {}
+
+
+def init(data_exchange_path, data_exchange_format="pytorch"):
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_common/model_exchange/__init__.py]: changed as suggested.

File: examples/advanced/loose-integration/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Loose Integration with NVFlare
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/loose-integration/README.md]: I was going to add ORNL, AV, and all cases here.
But we can narrow them down, so I change the dir name to "ml to fl"

File: examples/advanced/ml-to-fl/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# ML to FL trasition with NVFlare
+
+In this directory, we are providing job configurations to showcase how to utilizes 
+`LauncherExecutor`, `Launcher` and several NVFlare interfaces to simplify the
+transition from your ML code to FL with NVFlare.
+
+
+## Examples
+
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply examples/advanced/ml-to-fl/README.md:23]: Added more text.

File: examples/advanced/ml-to-fl/jobs/client_api/app/custom/cifar10.py [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+trainloader = torch.utils.data.DataLoader(trainset, batch_size=batch_size, shuffle=True, num_workers=2)
+
+testset = torchvision.datasets.CIFAR10(root=DATASET_PATH, train=False, download=True, transform=transform)
+testloader = torch.utils.data.DataLoader(testset, batch_size=batch_size, shuffle=False, num_workers=2)
+
+
+net = Net()
+
+# initializes NVFlare interface
+flare.init(config="config/config_exchange.json")
```
Comment (YuanTingHsieh) [inline examples/advanced/ml-to-fl/jobs/client_api/app/custom/cifar10.py:40]: default to "config/config_exchange.json"

File: nvflare/app_opt/pt/file_pipe_launcher_executor.py [hunk 8]
```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
```
[the remark this answers is not in the document]
Comment (YuanTingHsieh) [reply nvflare/app_opt/pt/file_pipe_launcher_executor.py:1]: to be consistent with other files inside nvflare/app_opt/pt, let's keep this name

File: examples/advanced/ml-to-fl/jobs/client_api1/app/custom/cifar10.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+import torchvision.transforms as transforms
+from net import Net
+
+# (1.0) import nvflare client API
+import nvflare.client as flare
+
+# (optional) set a fix place so we don't need to download everytime
+DATASET_PATH = "/tmp/nvflare/data"
+# (optional) use GPU to speed things up
+device = "cuda:0"
```
Comment (YuanTingHsieh) [inline examples/advanced/ml-to-fl/jobs/client_api1/app/custom/cifar10.py:28]: if want cpu, comment out
